detection/detection.go

```diff
@@ -0,0 +1,720 @@
+	result.OverallConfidence = calculateOverallConfidence(result.Details)
+	
+	// Log detection performance metrics
+	log.Printf("Object detection completed in %dms with overall confidence %.2f", 
+		result.LatencyMs, result.OverallConfidence)
+	
+	return result, nil
+}
+
+// DetectionConfig holds configuration for object detection
+type DetectionConfig struct {
+	UseOpenCV bool          // Whether to try OpenCV detection first
+	Timeout   time.Duration // Maximum time for detection
+}
+
+// analyzeWithEnhancedDetection performs the original enhanced detection as fallback
+func analyzeWithEnhancedDetection(photoPath string) (*DetectionResult, error) {
+	// Open and decode the image
+	file, err := os.Open(photoPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open photo: %v", err)
+	}
+	defer file.Close()
+
+	img, format, err := image.Decode(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode image: %v", err)
+	}
+
+	if format != "jpeg" && format != "png" && format != "gif" {
+		log.Printf("Warning: image format %s may not be fully supported, continuing with enhanced analysis", format)
+	}
+
+	// Initialize result
+	result := &DetectionResult{
+		PhotoPath:         photoPath,
+		Objects:           []string{},
+		Details:           []ObjectDetail{},
+		LatencyMs:         0, // Will be set by calling function
+		OverallConfidence: 0, // Will be calculated by calling function
+	}
+
+	// Analyze time of day using brightness
+	result.IsDay = analyzeTimeOfDayEnhanced(img)
+
+	// Perform enhanced object detection
+	objects, details := detectObjectsEnhanced(img)
+	result.Objects = objects
+	result.Details = details
+
+	// Generate summary
+	result.Summary = generateSummary(result.IsDay, objects)
+
+	return result, nil
+}
+
+// analyzeTimeOfDayEnhanced determines if it's day or night using enhanced brightness analysis
+func analyzeTimeOfDayEnhanced(img image.Image) bool {
+	bounds := img.Bounds()
+	width := bounds.Max.X - bounds.Min.X
+	height := bounds.Max.Y - bounds.Min.Y
+	
+	if width == 0 || height == 0 {
+		return true // default to day if we can't analyze
+	}
+
+	// Enhanced sampling with multiple metrics
+	sampleSize := 20 // More intensive sampling for better accuracy
+	totalBrightness := uint64(0)
+	totalContrast := uint64(0)
+	sampleCount := 0
+
+	for y := bounds.Min.Y; y < bounds.Max.Y; y += sampleSize {
+		for x := bounds.Min.X; x < bounds.Max.X; x += sampleSize {
+			r, g, b, _ := img.At(x, y).RGBA()
+			// Convert to 8-bit and calculate brightness using standard formula
+			brightness := (0.299*float64(r>>8) + 0.587*float64(g>>8) + 0.114*float64(b>>8))
+			totalBrightness += uint64(brightness)
+			
+			// Calculate local contrast by comparing with nearby pixels
+			if x+sampleSize < bounds.Max.X && y+sampleSize < bounds.Max.Y {
+				r2, g2, b2, _ := img.At(x+sampleSize, y+sampleSize).RGBA()
+				brightness2 := (0.299*float64(r2>>8) + 0.587*float64(g2>>8) + 0.114*float64(b2>>8))
```

Comment:
move the magic numbers (e.g. `0.299`) into named constants to improve documentation re: why were those values chosen and what formula is this referencing